File: app/src/main/java/in/completecourse/questionbank/adapter/NewArrivalAdapter.java
package in.completecourse.questionbank.adapter;

import android.content.Context;
import android.view.GestureDetector;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import java.util.List;
import in.completecourse.questionbank.model.BookNewArrival;
import in.completecourse.questionbank.R;

/**
 * RecyclerView adapter class to render items
 * This class can go into another separate class, but for simplicity
 */
public class NewArrivalAdapter extends RecyclerView.Adapter<NewArrivalAdapter.MyViewHolder> {

    private final List<BookNewArrival> booklist;
    private final Context context;

    class MyViewHolder extends RecyclerView.ViewHolder {
        final TextView name;
        final TextView price;
        final TextView code;
        final ImageView thumbnail;
        //final TextView rupeeSign;

        MyViewHolder(View view) {
            super(view);
            name = view.findViewById(R.id.title);
            price = view.findViewById(R.id.price);
            code = view.findViewById(R.id.code_store);
            thumbnail = view.findViewById(R.id.thumbnail);
            //rupeeSign = view.findViewById(R.id.ruppee_sign);
        }
    }

    public NewArrivalAdapter(Context context, List<BookNewArrival> booklist) {
        this.booklist = booklist;
        this.context = context;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.new_arrival_item_row, parent, false);
        return new MyViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, final int position) {
        final BookNewArrival book = booklist.get(position);
        holder.name.setText(book.getTitle());
        holder.price.setText(book.getRate());
        holder.code.setText(book.getCode());

        Glide.with((context))
                .asBitmap()
                .load(book.getUrl())
                .placeholder(R.drawable.background_gradient)
                .into(holder.thumbnail);
    }

    @Override
    public int getItemCount() {
        return booklist.size();
    }

    public interface ClickListener {
        void onClick(int position);
    }

    public static class RecyclerTouchListener implements RecyclerView.OnItemTouchListener {
        private final GestureDetector gestureDetector;

        private final NewArrivalAdapter.ClickListener clickListener;
        public RecyclerTouchListener(Context context, final NewArrivalAdapter.ClickListener clickListener){
            this.clickListener = clickListener;
            gestureDetector = new GestureDetector(context, new GestureDetector.SimpleOnGestureListener(){
                @Override
                public boolean onSingleTapUp(MotionEvent e) {
                    return true;
                }

            });
        }

        @Override
        public boolean onInterceptTouchEvent(@NonNull RecyclerView recyclerView, @NonNull MotionEvent motionEvent) {
            View child = recyclerView.findChildViewUnder(motionEvent.getX(),motionEvent.getY());
            if (child!=null && clickListener != null && gestureDetector.onTouchEvent(motionEvent)){
                clickListener.onClick(recyclerView.getChildPosition(child));
            }
            return false;
        }

        @Override
        public void onTouchEvent(@NonNull RecyclerView recyclerView, @NonNull MotionEvent motionEvent) {

        }

        @Override
        public void onRequestDisallowInterceptTouchEvent(boolean b) {

        }
    }
}

File: app/src/main/java/in/completecourse/questionbank/adapter/NewArrivalAdapter.kt
package `in`.completecourse.questionbank.adapter

import `in`.completecourse.questionbank.R
import `in`.completecourse.questionbank.model.BookNewArrival
import android.content.Context
import android.view.*
import android.view.GestureDetector.SimpleOnGestureListener
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.OnItemTouchListener
import com.bumptech.glide.Glide

/**
 * RecyclerView adapter class to render items
 * This class can go into another separate class, but for simplicity
 */
class NewArrivalAdapter(context: Context, booklist: List<BookNewArrival>) : RecyclerView.Adapter<NewArrivalAdapter.MyViewHolder>() {
    private val booklist: List<BookNewArrival> = booklist
    private val context: Context = context

    inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val name: TextView = view.findViewById(R.id.title)
        val price: TextView = view.findViewById(R.id.price)
        val code: TextView = view.findViewById(R.id.code_store)
        val thumbnail: ImageView = view.findViewById(R.id.thumbnail)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.new_arrival_item_row, parent, false)
        return MyViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val book: BookNewArrival = booklist[position]
        holder.name.text = book.title
        holder.price.text = book.rate
        holder.code.text = book.code
        Glide.with(context)
                .asBitmap()
                .load(book.url)
                .placeholder(R.drawable.background_gradient)
                .into(holder.thumbnail)
    }

    override fun getItemCount(): Int {
        return booklist.size
    }

    interface ClickListener {
        fun onClick(position: Int)
    }

    class RecyclerTouchListener(context: Context?, private val clickListener: ClickListener?) : OnItemTouchListener {
        private val gestureDetector: GestureDetector = GestureDetector(context, object : SimpleOnGestureListener() {
            override fun onSingleTapUp(e: MotionEvent): Boolean {
                return true
            }
        })

        override fun onInterceptTouchEvent(recyclerView: RecyclerView, motionEvent: MotionEvent): Boolean {
            val child = recyclerView.findChildViewUnder(motionEvent.x, motionEvent.y)
            if (child != null && clickListener != null && gestureDetector.onTouchEvent(motionEvent)) {
                clickListener.onClick(recyclerView.getChildPosition(child))
            }
            return false
        }

        override fun onTouchEvent(recyclerView: RecyclerView, motionEvent: MotionEvent) {}
        override fun onRequestDisallowInterceptTouchEvent(b: Boolean) {}

    }

}